lint/package.go

```diff
@@ -182,17 +183,41 @@ func (p *Package) scanSortable() {
 	}
 }
 
-func (p *Package) lint(rules []Rule, config Config, failures chan Failure) {
+func (p *Package) lint(rules []Rule, config Config, failures chan Failure) error {
 	p.scanSortable()
 	var wg sync.WaitGroup
+	errCh := make(chan error, 1) // Buffered channel to signal the first error
+    _, cancel := context.WithCancel(context.Background())
+
+    defer cancel() // Ensure the context is canceled at the end
```

Comment:
Using a cancelable context only make sense if you use a context.

So you should remove this
```suggestion
```
And [this](https://github.com/mgechev/revive/pull/1126#discussion_r1844984529)